rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java

```diff
@@ -83,34 +84,23 @@ protected T executeAutoCommitFalse(Object[] args) throws Exception {
      * @throws Throwable the throwable
      */
     protected T executeAutoCommitTrue(Object[] args) throws Throwable {
-        T result = null;
         AbstractConnectionProxy connectionProxy = statementProxy.getConnectionProxy();
-        LockRetryController lockRetryController = new LockRetryController();
         try {
             connectionProxy.setAutoCommit(false);
-            while (true) {
-                try {
-                    result = executeAutoCommitFalse(args);
-                    connectionProxy.commit();
-                    break;
-                } catch (LockConflictException lockConflict) {
-                    connectionProxy.getTargetConnection().rollback();
-                    lockRetryController.sleep(lockConflict);
-                } catch (Exception exx) {
-                    connectionProxy.getTargetConnection().rollback();
-                    throw exx;
-                }
-            }
-
+            return new LockRetryPolicy(connectionProxy.getTargetConnection()).execute(() -> {
+                T result = executeAutoCommitFalse(args);
+                connectionProxy.commit();
+                return result;
+            });
         } catch (Exception e) {
             // when exception occur in finally,this exception will lost, so just print it here
             LOGGER.error("execute executeAutoCommitTrue error:{}", e.getMessage(), e);
+            connectionProxy.rollback();
```

Comment:
just used for LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT=false

Reply:
Should it be okay to do `ConnectionProxy#rollback` in both cases?

Reply:
> Should it be okay to do `ConnectionProxy#rollback` in both cases?

there is no problem. I just mark ,when   LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT=true ,it  already rollback , never run here.